src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -461,8 +461,16 @@ private String getPrincipalName(String username, String domainName) {
                             throw e;
                         }
                     }
-                    if (!found || duration >= 10) {
-                        LOGGER.warning(String.format("AD chain lookup is taking too long (%dms). Falling back to recursive lookup", duration));
+                    if (!found && duration >= 10) {
+                        LOGGER.warning(String.format("AD chain didn't find the group in (%dms). Falling back to recursive lookup", duration));
+                        groupLookupStrategy = GroupLookupStrategy.RECURSIVE;
+                        continue;
+                    } else if (found && duration >= 10) {
+                        LOGGER.warning(String.format("AD chain lookup found the group but took too long (%dms). Falling back to recursive lookup", duration));
+                        groupLookupStrategy = GroupLookupStrategy.RECURSIVE;
+                        return groups;
+                    } else if (!found) {
+                        LOGGER.warning(String.format("AD chain lookup didn't found the group. Falling back to recursive lookup", duration));
```

Comment:
Nit: the English could be clearer.
"Group lookup via Active Directory\'s 'matching rule in chain' extension failed. Falling back to recursive group lookup strategy for this and future queries"